test/e2e/nodeobservabiltiyrun_test.go

```diff
@@ -35,6 +44,101 @@ import (
 //var k8sClient client.Client
 //var testEnv *envtest.Environment
 
+const (
+	nodeobservability = "nodeobservability-sample"
+	image             = "quay.io/josefkarasek/node-observability-agent:latest"
+)
+
+func TestAgentAvailable(t *testing.T) {
+	if err := waitForDaemonsetStatus(t, kubeClient); err != nil {
+		t.Errorf("Did not get expected available condition: %v", err)
+	}
+}
+func TestNodeObservabilityRun(t *testing.T) {
+	var (
+		err error
+	)
+	if err := v1alpha1.AddToScheme(Scheme); err != nil {
+		panic(err)
+	}
+	if err = initKubeClient(); err != nil {
+		fmt.Printf("Failed to create kube client: %v\n", err)
+		os.Exit(1)
+	}
+	if err = ensureNodeObservabilityResource(); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to create NodeObservability Resource in ns %s: %v\n", operandNamespace, err)
+	}
+	nodeObsRun := ensureNodeObservabilityRunResource()
+	if err = kubeClient.Create(context.TODO(), nodeObsRun); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to create NodeObservabilityRun Resource in ns %s: %v\n", operandNamespace, err)
+	}
+}
+func ensureNodeObservabilityResource() error {
+	spec := v1alpha1.NodeObservabilitySpec{
+		Labels: map[string]string{},
+		Image:  image,
+	}
+
+	nodeObs := v1alpha1.NodeObservability{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodeobservability,
+			Namespace: operandNamespace,
+		},
+		Spec: spec,
+	}
+	return kubeClient.Create(context.TODO(), &nodeObs)
+}
+func ensureNodeObservabilityRunResource() *v1alpha1.NodeObservabilityRun {
+	spec := v1alpha1.NodeObservabilityRunSpec{
+		NodeObservabilityRef: &v1alpha1.NodeObservabilityRef{
+			Name: nodeobservability,
+		},
+	}
+
+	nodeObs := v1alpha1.NodeObservabilityRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodeobservability,
+			Namespace: operandNamespace,
+		},
+		Spec: spec,
+	}
+	return &nodeObs
+}
+func TestIfNodeObservabilityRunFinished(t *testing.T) {
+	if err := IfFinished(t); err != nil {
+		t.Errorf("Node Observability Run is Failed: %v", err)
+	}
+}
+func IfFinished(t *testing.T) error {
+	t.Helper()
+	return wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		nodeObsRun := ensureNodeObservabilityRunResource()
+		err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: nodeobservability, Namespace: operandNamespace}, nodeObsRun)
+		if err != nil {
+			t.Logf("Failed to get NodeObservabilityRun Resource in ns %s: %v\n", operandNamespace, err)
+			return false, nil
+		}
+		isFinished := nodeObsRun.Status.FinishedTimestamp
+		if isFinished != nil && !isFinished.IsZero() && len(nodeObsRun.Status.FailedAgents) == 0 {
+			fmt.Printf("Node Observability Run is Finished Successfully")
```

Comment:
nit: Change Node Observability Run is Finished Successfully -> Node Observability Run has Finished Successfully